ksql-engine/src/main/java/io/confluent/ksql/codegen/SqlToJavaVisitor.java

```diff
@@ -418,12 +420,15 @@ private String visitBooleanComparisonExpression(final ComparisonExpression.Type
       return new Pair<>(expr, Schema.OPTIONAL_BOOLEAN_SCHEMA);
     }
 
+    // CHECKSTYLE_RULES.OFF: CyclomaticComplexity
```

Comment:
Noooooo........ this rule is generally a clear indication the logic of the function is too complex to easily grok.  Don't disable the rule, instead refactor, e.g. move your decimal handling into `visitDecimalCast`, or better still replace the switch with code that looks up the handler from a static map.

`SqlToJavaVisitor` is a bit of a god class - we need to think about how to break it up.  However in the mean time, maybe handle the cast functionality in a utility class, and just call that from here.  This encourages people to break up the functionality into small targeted methods that are easy to understand... and improves test coverage!

Reply:
> e.g. move your decimal handling into visitDecimalCast

did this I guess after you had written your comment :) but it's still not good enough for checkstyle

> or better still replace the switch with code that looks up the handler from a static map.

okay you caught me being lazy

Reply:
Yup, I did. =)

I reckon the best approach is to pull the full function out into a util class and then refactor it and test it.  but that is more work...

Reply:
Testing it in isolation is a little weird - I just refactored it into an internal static class and used a handler map